File: relisoft-hr/server/models/Team.js
import mongoose from 'mongoose';

const teamSchema = new mongoose.Schema({
  name: { type: String, required: true, trim: true },
  project: { type: mongoose.Schema.Types.ObjectId, ref: 'Project', required: true },
  lead: { type: mongoose.Schema.Types.ObjectId, ref: 'Employee' },
}, { timestamps: true });

teamSchema.index({ name: 1, project: 1 }, { unique: true });

export default mongoose.model('Team', teamSchema);

File: relisoft-hr/server/Models/Team.cs
using System.ComponentModel.DataAnnotations;

namespace RelisoftHR.Models;

public class Team : IHasRowVersion
{
    [Key]
    public int Id { get; set; }
    [Required, MaxLength(200)]
    public string Name { get; set; } = "";
    public int ProjectId { get; set; }
    public int LeadId { get; set; }
    public DateTime CreatedOn { get; set; } = DateTime.UtcNow;
    public byte[]? RowVersion { get; set; }

    public Project? Project { get; set; }
    public Employee? Lead { get; set; }
    public ICollection<EmployeeTeam> EmployeeTeams { get; set; } = new List<EmployeeTeam>();
}
